apps/balancer-tools/src/app/poolsimulator/(components)/SwapCurve.tsx

```diff
@@ -317,3 +364,60 @@ export function SwapCurve() {
     />
   );
 }
+
+function getBetaLimitIndexes({
+  analysisTokenOut,
+  analysisTokenIn,
+  tabTokenIn,
+  tabTokenOut,
+  analysisTokenInitialBalance,
+  tabTokenInitialBalance,
+  beta,
+}: {
+  analysisTokenOut: number[];
+  analysisTokenIn: number[];
+  tabTokenIn: number[];
+  tabTokenOut: number[];
+  analysisTokenInitialBalance: number;
+  tabTokenInitialBalance: number;
+  beta: number;
+}) {
+  const analysisTokenSwapAmounts = [...analysisTokenOut]
+    .reverse()
+    .concat(analysisTokenIn);
+
+  const tabTokenSwapAmounts = [...tabTokenIn].reverse().concat(tabTokenOut);
+
+  const analysisTokenBalanceAfterSwap = analysisTokenSwapAmounts.map(
+    (amount) => amount + analysisTokenInitialBalance,
+  );
+  const tabTokenBalanceAfterSwap = tabTokenSwapAmounts.map(
+    (amount) => amount + tabTokenInitialBalance,
+  );
+
+  const isBalancesInBetaRegion = analysisTokenBalanceAfterSwap.map(
+    (balanceX, index) => {
+      const balanceY = tabTokenBalanceAfterSwap[index];
+      return Math.abs(balanceX - balanceY) / balanceX + balanceY < beta;
+    },
+  );
+
+  const betaLimitIndexes = [];
+  for (let i = 1; i < isBalancesInBetaRegion.length; i++) {
+    if (isBalancesInBetaRegion[i] !== isBalancesInBetaRegion[i - 1]) {
+      betaLimitIndexes.push(i);
+    }
+    if (betaLimitIndexes.length === 2) break;
```

Comment:
could there be more than 2? 👀

Reply:
I'll add a comment about this, but it shouldn't have more than 2 bc the swap amount derivative should be always negative.